src/RemoteInboxNotifications/GetRuleProcessor.php

```diff
@@ -59,6 +59,10 @@ public static function get_processor( $rule_type ) {
 				);
 			case 'data':
 				return new DataRuleProcessor();
+			case 'order_count':
+				return new OrderCountRuleProcessor(
+					new OrdersProvider()
```

Comment:
Nit: what do you think about having this occur by default in the `OrderCountRuleProcessor` constructor (so we only pass in an override for testing)?

Reply:
I don't think it's a bad idea, I just have an aversion to default parameters because I've seen code get burned by them in the past - having to debug a bad assumption with a default parameter can be difficult

Reply:
I've ended up doing this anyway since it was suggested in another PR as well...